07_classes.md

```diff
@@ -0,0 +1,565 @@
+
+There are multiple reasons classes are useful. They will help you to follow some of the principles of "good" object-oriented programming, such as the five "SOLID" principles, which are covered in detail in module 2 of the course. But here are two fundamental uses of classes:
+
+### Group variables together into entities
+
+You don't want to pass around multiple variables (like `dog_name`, `owner`, `breed` etc) for each dog that passes through the system. So group them together into a single entity (a `Dog`). Data classes are simple classes that hold data and don't necessarily provide any functionality beyond that.
+
+This _could_ be achieved with a dictionary but a class is more robust. The class definition is a single source of truth for what it means to be a Dog. You can ensure that every Dog in your system has the same shape. Dictionaries are less reliable - key/value pairs can be added/removed at any point and it's not obvious where to look to find out what your dictionary must/can contain.
+
+You can also for example make some of the dog's data constant or optional, without causing bugs because it's clearly laid out in the definition of Dog.
+
+### Linking functionality with state
+
+In Python, you could put all of your dog-related functions in a "dog_functions.py" module. But putting it into the Dog class could be nicer. E.g. say you want a function `is_healthy_weight` that tells you whether an individual dog is healthy based on the data you hold about it. Putting it in a module means you need something like this:
+
+```python
+import dog_functions
+
+my_dog_is_healthy = dog_functions.is_healthy_weight(my_dog)
+```
+
+If the function is part of the Dog class, then using it looks like this:
+
+```python
+my_dog_is_healthy = my_dog.is_healthy_weight()
+```
+
+## Defining a class
+
+To define a class in Python, write the keyword `class` followed by a name for your new class. Put the following code in a new file. Here we will use the `pass` keyword to say that it's deliberately empty, similar to how you would define an empty function.
+
+```python
+class Dog:
+    pass
+```
+
+> In a real project you will often put a class in its own module (i.e. file) in order to keep the project organised and easier to navigate. You then import the class in another file to actually use it. E.g. `from dog import Dog` if you put it in a file called dog.py. These exercises won't ask you to do so but feel free to try it out for practice.
+
+You create an instance of the class by calling the class like a function - i.e. adding parentheses `()`. Add this code to your file, under the class definition:
+
+```python
+my_dog = Dog()
+print(Dog)
+print(my_dog)
+``` 
+
+You should see that Dog is a clasa and as for my_dog - 
+we haven't defined a nice text representation for Dogs, so it will print out something like `<__main__.Dog object at 0x000001F6CBBE7BB0>`. 
```

Comment:
Typo `clasa` as Jack said. Also maybe include the output for `print(Dog)` too? (`<class '__main__.Dog'>`)